PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/messages/AzureBundle.java

```diff
@@ -33,11 +36,27 @@ public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) Stri
         return ourInstance.getMessage(key, params);
     }
 
+    public static IAzureOperationTitle operation(@NotNull @PropertyKey(resourceBundle = BUNDLE) String name, @NotNull Object... params) {
+        return MessageBundleBasedOperationTitle.builder().name(name).params(params).build();
+    }
+
     @NonNls
     private static final String BUNDLE = "com.microsoft.intellij.ui.messages.messages";
     private static final AzureBundle ourInstance = new AzureBundle();
 
     private AzureBundle() {
         super(BUNDLE);
     }
+
+    @Builder
+    @Getter
+    public static class MessageBundleBasedOperationTitle implements IAzureOperationTitle {
+        private final String name;
+        private final Object[] params;
+
+        @Override
+        public String getTitle() {
+            return AzureBundle.ourInstance.getMessage(this.name, params);
+        }
+    }
```

Comment:
interface should be for function or capacity. is it not enough if only let the class as POJO without implementing the interface?

Reply:
> interface should be for function or capacity
it's not true. e.g. `java.util.List`/`java.util.Collection`

> is it not enough if only let the class as POJO
no, it's not enough, since, the implementation of `getTitle` may vary to cases.